modules/metadata-support/src/main/java/org/mule/runtime/metadata/internal/cache/ComponentBasedIdHelper.java

```diff
@@ -69,6 +86,18 @@ public static Optional<String> resolveConfigName(ComponentAst elementModel) {
         .map(param -> param.getResolvedRawValue());
   }
 
+  public static List<String> parameterNamesRequiredForMetadataCacheId(ComponentAst component) {
+    return component.getModel(EnrichableModel.class)
+        .flatMap(model -> model.getModelProperty(RequiredForMetadataModelProperty.class)
+            .map(RequiredForMetadataModelProperty::getRequiredParameters))
+        .orElse(emptyList());
+  }
+
+  public static MetadataCacheId resolveDslTagId(ComponentAst elementModel) {
```

Comment:
`resolveDslTagId` -> `resolveComponentIdentifierMetadataCacheId`, this will not be tied only to a tag-based dsl